spicy/toolchain/src/compiler/codegen/parsers/literals.cc

```diff
@@ -50,9 +50,7 @@ struct Visitor : public visitor::PreOrder {
     auto context() { return pb()->context(); }
     auto pushBuilder(std::shared_ptr<Builder> b) { return pb()->pushBuilder(std::move(b)); }
     auto pushBuilder() { return pb()->pushBuilder(); }
-    auto pushBuilder(std::shared_ptr<Builder> b, const std::function<void()>& func) {
-        return pb()->pushBuilder(std::move(b), func);
-    }
+    auto pushBuilder(std::shared_ptr<Builder> b, void (*func)()) { return pb()->pushBuilder(std::move(b), func); }
```

Comment:
Are you deliberately ruling out callbacks with state here? I think passing a lambda (with `T&&`) would be more flexible and in line with similar APIs elsewhere.

Reply:
Did it because we don't need stateful so far and it allows getting away without a template. But I agree with the consistency argument, so changing to `T&&`.